jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy

```diff
@@ -11,15 +14,21 @@ import org.gradle.api.Project
  */
 class JRubyPlugin implements Plugin<Project> {
     static final String TASK_GROUP_NAME = 'JRuby'
+    static final String EMBEDDED_SERVER_TASK = 'startEmbeddedGemArtifactsServer'
 
     static final String RUBYGEMS_RELEASE_URL = 'http://rubygems.lasagna.io/proxy/maven/releases'
 
     void apply(Project project) {
+        project.gradle.startParameter.taskNames = [EMBEDDED_SERVER_TASK] + project.gradle.startParameter.taskNames
         project.extensions.create('jruby', JRubyPluginExtension, project)
+        setupRubygemsRepositories(project)
+        setupRubygemsRelease(project)
 
-        if (!project.repositories.metaClass.respondsTo(project.repositories, 'rubygemsRelease')) {
-            project.repositories.metaClass.rubygemsRelease << { ->
-                maven { url RUBYGEMS_RELEASE_URL }
+        // put the embedded server into task due to no classloader problems
+        project.tasks.create(name: EMBEDDED_SERVER_TASK) {
```

Comment:
I'm confused with how the task is necessary? Task execution would come after dependency resolution if I remember correctly, so I'm not quite understanding how this would end up starting the server in time to resolve dependencies